Add additional init_argument to register method

LucasG2000 opened the pull request:
Sometimes it is needed to pass additional information to the inherited FlaskView classes.

As flask-classy and flask-classful are defined, the __init__ call is
intended to being created without arguments, because when instanced it
is created by just a call without parameters.

While this can be overcomed referencing the needed information from
global objects, this practice is not as clean as clearly defining and
passing the information through the constructor.

This commit adds additional init_argument to register method, optional
with default value of None, that allows to pass information in the
__init__ of the registered classes.

The implementation is fully backwards compatible, with an optional
parameter than when not provided will behave just like it before.

This maybe solves the request in teracyhq/flask-classful/issues#84

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/teracyhq/flask-classful?pullRequest=86) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/teracyhq/flask-classful?pullRequest=86) before we can accept your contribution.<br/><hr/>**Administrator** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/teracyhq/flask-classful?pullRequest=86) it.</sub>

hoatle commented:
@LucasG2000 could you add unit test for this? and then we need to update the docs for this change, too.

LucasG2000 commented:
Sure, I will commit something tomorrow.

LucasG2000 commented:
Done, added unit tests and changed documentation index.rst.

Note that while unit tests have been checked, I had to do changes in index.rst blindly, while following the instructions on how to work with the documentation I'm getting errors in  `docker-compose -f docker-compose.prod.yml up` like:

```
vagrant@teracy:~/workspace/flask-classful/docs$ docker-compose -f docker-compose.prod.yml up
WARNING: The CI_BUILD_ID variable is not set. Defaulting to a blank string.
WARNING: The CI_BUILD_REF variable is not set. Defaulting to a blank string.
WARNING: The CI_BUILD_TIME variable is not set. Defaulting to a blank string.
WARNING: The CI_REGISTRY_IMAGE variable is not set. Defaulting to a blank string.
WARNING: Found orphan containers (docs_app-dev_1) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up.
Recreating docs_app-prod_1 ... done
Creating docs_ngrok-prod_1 ... error

ERROR: for docs_ngrok-prod_1  Cannot create container for service ngrok-prod: could not get container for nginx-proxy: No such container: nginx-proxy

ERROR: for ngrok-prod  Cannot create container for service ngrok-prod: could not get container for nginx-proxy: No such container: nginx-proxy
ERROR: Encountered errors while bringing up the project.
```

Nevertheless don't worry, I think the changes in the doc are ok.
